src/dotty/tools/dotc/reporting/Reporter.scala

```diff
@@ -236,7 +240,7 @@ abstract class Reporter extends interfaces.ReporterResult {
     override def default(key: String) = 0
   }
 
-  def report(d: MessageContainer)(implicit ctx: Context): Unit =
+  def report(d: => MessageContainer)(implicit ctx: Context): Unit =
```

Comment:
1. `MessageContainer` still takes messages by-name so I doubt it needs to be by-name itself.
2. If there's some reason I'm missing (totally possible), other uses of `MessageContainer` would have to also become by-name—starting from `isHidden`. A test might be harder, so one would have to grep for uses of `MessageContainer`.